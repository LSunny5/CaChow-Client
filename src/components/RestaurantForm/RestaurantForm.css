/*edit page*/
.RFormContent {
    min-height: 91vh;
    padding: 1rem;
}

/*form container*/
.formBox {
    text-align: left;
    width: 95%;
    margin: 0 auto;
    background: white;
    padding: 1rem 0.5rem;
    border-radius: 30px
}

/*labels for fields*/
.inputLabel  {
    font-weight: bold;
    font-size: 1.2rem;
    font-family: 'Cinzel', serif;
}


/*fields for inputs*/
.textField {
    margin: 10px;
    font-size: 1rem;
    font-family: 'Delius', cursive;
    width: 90%;
    border: none;
    border-bottom: 2px solid black; 
    padding: 0.5rem;
}

/*container for buttons*/
.buttonBox {
    text-align: center;
    display:flex;
    flex-direction: row;
    justify-content: space-evenly;
    align-items: center;
    margin: 1rem auto 0;
}

/*button look*/
.generalButton {
    display: inline-block;
    text-decoration: none;
    border-radius: 5px;
    border: none;
    font-family: 'Delius', cursive;
    margin: 0.3rem;
    width: 40%;
    padding: 0.7rem 1.5rem;
    outline: none;
    font-size: 1.25rem;
    color: white;
    line-height: 1.5rem;
    font-weight: 400;
    cursor: pointer;
    transition: all 150ms ease-out;
}

/*update, add button*/
.submit {
    background-color:green;
    transition-duration: 75ms;
    box-shadow: 0 0 0 0.2rem green;
}

/*submit hover*/
.submit:hover {
    box-shadow: 0 0 0 0.2rem white, 0 0 0 0.4rem green;
}

/*onclick update add button*/
.submit:focus {
    box-shadow: 0 0 0 0.4rem green;
}

/*cancel button*/
.cancel {
    background-color:red;
    transition-duration: 75ms;
    box-shadow: 0 0 0 0.2rem red;
}

/*cancel hover*/
.cancel:hover {
    box-shadow: 0 0 0 0.2rem white, 0 0 0 0.4rem red;
}

/*onclick cancel button*/
.cancel:focus {
    box-shadow: 0 0 0 0.4rem red;
}

/********************************************************************/
/*For Ipads and larger screens */
@media only screen and (min-width: 600px) {
    /*form container*/
    .formBox {
        width: 85%;
    }

    /*container for buttons*/
    .buttonBox {
        justify-content: space-between;
        width: 80%;
    }

    /*button look*/
    .generalButton {
        margin: 0.5rem;
        width: 35%;
        padding: 1rem 1.5rem;
    }
}

/********************************************************************/
/*For Laptops and larger screens */
@media only screen and (min-width: 1024px) {
    /*form container*/
    .formBox {
        width: 75%;
        padding: 1rem;
    }

    .textField {
        width: 95%;
    }

    /*container for buttons*/
    .buttonBox {
        width: 60%;
    }
}