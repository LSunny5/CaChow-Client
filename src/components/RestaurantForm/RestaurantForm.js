import React from 'react';
import './RestaurantForm.css';
import config from '../../config';
import CachowContext from '../../CachowContext';
import TokenService from '../../services/token-service';

class RestaurantForm extends React.Component {
    static contextType = CachowContext;
    handleUpdate = event => {
        const { title, value } = event.target;
        this.setState({ [title]: value.trim() });
    }

    handleSubmit = (event) => {
        event.preventDefault();
        let uname = localStorage.getItem('name');
        let lastH = this.context.hours[this.context.hours.length - 1];

        const newRestaurant = {
            r_owner: uname,
            r_image: event.target['restaurantImage'].value,
            r_type: event.target['restaurantType'].value,
            r_name: event.target['restaurantName'].value,
            r_address: event.target['restaurantAddress'].value,
            r_city: event.target['restaurantCity'].value,
            r_state: event.target['restaurantState'].value,
            r_zip: event.target['restaurantZip'].value,
            r_phone: event.target['restaurantPhone'].value,
            r_hours: lastH.hours_id,
        }

        fetch(`${config.APIENDPOINT}/restaurants`, {
            method: 'POST',
            headers: {
                'content-type': 'application/json',
                'authorization': `bearer ${TokenService.getAuthToken()}`,
            },
            body: JSON.stringify(newRestaurant),
        })
            .then(response => {
                if (!response.ok)
                    return response.json().then(e => Promise.reject(e))
                return response.json()
            })
            .then(restaurant => {
                this.context.addRestaurant(restaurant);
                let lastR = this.context.restaurants[this.context.restaurants.length - 1];
                alert('Restaurant successfully added!');
                window.location.href = `/restaurants/${lastR.r_id}/addItem`;
            })
            .catch(error => {
                console.error({ error })
                alert('Error! ' + error);
            })
    }

    handleCancel = (event) => {
        event.preventDefault();
        let lastH = this.context.hours[this.context.hours.length - 1];

        fetch(`${config.APIENDPOINT}/hours/${lastH.hours_id}`, {
            method: 'DELETE',
            headers: {
                'content-type': 'application/json',
                'authorization': `bearer ${TokenService.getAuthToken()}`,
            },
        })
            .then(response => {
                if (!response.ok)
                    return response.json().then(e => Promise.reject(e))
                return response
            })
            .then(rHours => {
                this.context.deleteHours(rHours.hours_id);
                alert('Adding Restaurant process cancelled, returning to account screen...');
                window.location.href = `/account`;
            })
            .catch(error => {
                console.error({ error })
                alert('Error! ' + error);
            })
    }

    render() {
        return (
            <form className="addForm" onSubmit={this.handleSubmit}>
                <fieldset>
                    <label htmlFor="restaurantName" className="inputLabel">Name of Restaurant: </label>
                    <input
                        type="text"
                        id="restaurantName"
                        name="restaurantName"
                        className="textField"
                        placeholder="Add name of restaurant here..."
                        onChange={this.handleUpdate}
                    />
                    <br />
                    <label htmlFor="restaurantImage" className="inputLabel">Add an image: </label>
                    <input
                        type="text"
                        id="restaurantImage"
                        name="restaurantImage"
                        className="textField"
                        placeholder="Add url of image here..."
                        onChange={this.handleUpdate}
                    />
                    <br />
                    <label htmlFor="restaurantType" className="inputLabel">What type of restaurant? </label>
                    <input
                        type="text"
                        id="restaurantType"
                        name="restaurantType"
                        className="textField"
                        placeholder="Add type here..."
                        onChange={this.handleUpdate}
                    />
                    <br />
                    <label htmlFor="restaurantAddress" className="inputLabel">Address: </label>
                    <input
                        type="text"
                        id="restaurantAddress"
                        name="restaurantAddress"
                        className="textField"
                        placeholder="Add address here..."
                        onChange={this.handleUpdate}
                    />
                    <br />
                    <label htmlFor="restaurantCity" className="inputLabel">City: </label>
                    <input
                        type="text"
                        id="restaurantCity"
                        name="restaurantCity"
                        className="textField"
                        placeholder="Add city here..."
                        onChange={this.handleUpdate}
                    />
                    <br />
                    <label htmlFor="restaurantState" className="inputLabel">State: </label>
                    <input
                        type="text"
                        id="restaurantState"
                        name="restaurantState"
                        className="textField"
                        placeholder="Add state here..."
                        onChange={this.handleUpdate}
                    />
                    <br />
                    <label htmlFor="restaurantZip" className="inputLabel">ZipCode: </label>
                    <input
                        type="text"
                        id="restaurantZip"
                        name="restaurantZip"
                        className="textField"
                        placeholder="Enter zipcode "
                        onChange={this.handleUpdate}
                    />
                    <br />
                    <label htmlFor="restaurantPhone" className="inputLabel">Phone Number: </label>
                    <input
                        type="text"
                        id="restaurantPhone"
                        name="restaurantPhone"
                        className="textField"
                        placeholder="Add phone number here..."
                        onChange={this.handleUpdate}
                    />
                    <div className="buttonBox">
                        <button type='submit' className="button">
                            Next
                        </button>
                        <button onClick={this.handleCancel} className="button">
                            Cancel
                        </button>
                    </div>
                </fieldset>
            </form>
        );
    };
}

export default RestaurantForm;