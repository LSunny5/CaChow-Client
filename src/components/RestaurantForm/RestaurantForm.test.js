import React from 'react';
import ReactDOM from 'react-dom';
import RestaurantForm from './RestaurantForm';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { BrowserRouter } from 'react-router-dom';

describe(`RestaurantForm component`, () => {
    const props = {
        hours_id: 1, 
        sun_open: 'Open', 
        sun_close: 'Closed',
        mon_open: 'Open', 
        mon_close: 'Closed',
        tues_open: 'Open', 
        tues_close: 'Closed',
        wed_open: 'Open', 
        wed_close: 'Closed',
        thu_open: 'Open', 
        thu_close: 'Closed',
        fri_open: 'Open', 
        fri_close: 'Closed',
        sat_open: 'Open', 
        sat_close: 'Closed',
        hours_owner: 'owner 1',
    }

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <RestaurantForm {...props} />
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the RestaurantForm form', () => {
        const wrapper = shallow(<RestaurantForm {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
});