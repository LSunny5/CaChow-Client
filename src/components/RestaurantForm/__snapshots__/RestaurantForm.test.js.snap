// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantForm component renders the RestaurantForm form 1`] = `
<form
  className="addForm"
  onSubmit={[Function]}
>
  <fieldset>
    <label
      className="inputLabel"
      htmlFor="restaurantName"
    >
      Name of Restaurant: 
    </label>
    <input
      className="textField"
      id="restaurantName"
      name="restaurantName"
      onChange={[Function]}
      placeholder="Add name of restaurant here..."
      type="text"
    />
    <br />
    <label
      className="inputLabel"
      htmlFor="restaurantImage"
    >
      Add an image: 
    </label>
    <input
      className="textField"
      id="restaurantImage"
      name="restaurantImage"
      onChange={[Function]}
      placeholder="Add url of image here..."
      type="text"
    />
    <br />
    <label
      className="inputLabel"
      htmlFor="restaurantType"
    >
      What type of restaurant? 
    </label>
    <input
      className="textField"
      id="restaurantType"
      name="restaurantType"
      onChange={[Function]}
      placeholder="Add type here..."
      type="text"
    />
    <br />
    <label
      className="inputLabel"
      htmlFor="restaurantAddress"
    >
      Address: 
    </label>
    <input
      className="textField"
      id="restaurantAddress"
      name="restaurantAddress"
      onChange={[Function]}
      placeholder="Add address here..."
      type="text"
    />
    <br />
    <label
      className="inputLabel"
      htmlFor="restaurantCity"
    >
      City: 
    </label>
    <input
      className="textField"
      id="restaurantCity"
      name="restaurantCity"
      onChange={[Function]}
      placeholder="Add city here..."
      type="text"
    />
    <br />
    <label
      className="inputLabel"
      htmlFor="restaurantState"
    >
      State: 
    </label>
    <input
      className="textField"
      id="restaurantState"
      name="restaurantState"
      onChange={[Function]}
      placeholder="Add state here..."
      type="text"
    />
    <br />
    <label
      className="inputLabel"
      htmlFor="restaurantZip"
    >
      ZipCode: 
    </label>
    <input
      className="textField"
      id="restaurantZip"
      name="restaurantZip"
      onChange={[Function]}
      placeholder="Enter zipcode "
      type="text"
    />
    <br />
    <label
      className="inputLabel"
      htmlFor="restaurantPhone"
    >
      Phone Number: 
    </label>
    <input
      className="textField"
      id="restaurantPhone"
      name="restaurantPhone"
      onChange={[Function]}
      placeholder="Add phone number here..."
      type="text"
    />
    <div
      className="buttonBox"
    >
      <button
        className="button"
        type="submit"
      >
        Next
      </button>
      <button
        className="button"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </fieldset>
</form>
`;
